src/main/java/org/beanpod/switchboard/util/MaintainDeviceStatus.java

```diff
@@ -0,0 +1,93 @@
+package org.beanpod.switchboard.util;
+
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import lombok.RequiredArgsConstructor;
+import org.beanpod.switchboard.dao.DeviceDaoImpl;
+import org.beanpod.switchboard.dto.DecoderDto;
+import org.beanpod.switchboard.dto.EncoderDto;
+import org.beanpod.switchboard.dto.StreamDto;
+import org.beanpod.switchboard.dto.mapper.DeviceMapper;
+import org.beanpod.switchboard.entity.DummyInterface;
+import org.beanpod.switchboard.service.LogService;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class MaintainDeviceStatus {
+
+  private static final DateUtil date = new DateUtil();
+  private static final String ONLINE_STATUS = "online";
+  private static final String OFFLINE_STATUS = "offline";
+  private final DeviceDaoImpl service;
+  private final DeviceMapper deviceMapper;
+  private final LogService logService;
+
+  /*
+   * maintain and create logs when decoders or encoders are retrieved
+   * T should be of type DecoderEntity or EncoderEntity
+   */
+  public <T extends DummyInterface> void maintainStatusField(List<T> devices) {
+    Date dateToBeCompared = getDateToBeCompared();
+
+    for (int i = 0; i < devices.size(); i++) {
+      // if status field equals online and lastCommunication is more than 10minutes old
+      if (((devices.get(i)).getDevice().getStatus()).equalsIgnoreCase(ONLINE_STATUS)
+          && dateToBeCompared.after(devices.get(i).getLastCommunication())) {
+        // update last_communication field to offline
+        (devices.get(i).getDevice()).setStatus(OFFLINE_STATUS);
+        service.save(deviceMapper.toDeviceDto(devices.get(i).getDevice()));
+
+        // create a log
+        createLog(OFFLINE_STATUS, (devices.get(i)).getSerialNumber());
+      } else if (((devices.get(i)).getDevice().getStatus()).equalsIgnoreCase(OFFLINE_STATUS)
+          && dateToBeCompared.before(devices.get(i).getLastCommunication())) {
```

Comment:
i think devices.get(i).getdevice() is used multiple times in this code. maybe could be refactored into one line
or refactor devices.get(i) in one line

Reply:
Good suggestion. I've updated it.